app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -53,7 +53,7 @@ def study_modes(course)
     end
 
     def create_course_options(site, study_mode, site_status)
-      course_option = CourseOption.find_or_initialize_by(
+      course_option = CourseOption.find_or_create_by(
```

Comment:
What do you think about this change?

Reply:
Since we are updating attributes on `CourseOption` below, I would suggest we keep this as `find_or_initialize_by` to avoid writing to the record twice

Reply:
Could it be that valid course options are not created because of the `if course_option.vacancy_status != vacancy_status` conditional below? @raamSoftwire

Reply:
Yes Michael, I think you're correct. That conditional logic was added as a review markup and I hadn't considered that scenario. So I think `find_or_create_by` is what we want here.